lxd/storage/backend_lxd.go

```diff
@@ -5555,6 +5555,39 @@ func (b *lxdBackend) UpdateCustomVolume(projectName string, volName string, newD
 			}
 		}
 
+		sharedVolume, ok := changedConfig["security.shared"]
+		if ok && shared.IsFalseOrEmpty(sharedVolume) {
+			var usedByProfileDevices []api.Profile
+
+			err = VolumeUsedByProfileDevices(b.state, b.name, projectName, &curVol.StorageVolume, func(profileID int64, profile api.Profile, project api.Project, usedByDevices []string) error {
+				usedByProfileDevices = append(usedByProfileDevices, profile)
+
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+
+			if len(usedByProfileDevices) > 0 {
+				return fmt.Errorf("Cannot un-share custom storage block volume if attached to profile")
```

Comment:
Do you think it would be clearer to alter these errors (and any similar ones in this PR) to reference the `security.shared` setting, e.g. `Cannot disable security.shared on custom storage block volume as it is attached to profile(s)`

Reply:
Yes, I agree

Reply:
Yes I agree